src/Widgets/HeaderBar.vala

```diff
@@ -129,6 +166,43 @@ public class Camera.Widgets.HeaderBar : Gtk.HeaderBar {
         mode_switch.active = Camera.Application.settings.get_enum ("mode") == Utils.ActionType.VIDEO;
     }
 
+    public void add_camera_option (Gst.Device camera) {
+        var menuitem = new Gtk.MenuItem.with_label (camera.get_display_name ());
+
+        Gtk.CssProvider item_style_provider = new Gtk.CssProvider ();
+
+        try {
+            item_style_provider.load_from_data (CAMERA_MENU_BUTTON_STYLESHEET, -1);
+        } catch (Error e) {
+            warning ("Styling take button failed: %s", e.message);
+        }
+
+        var item_style_context = menuitem.get_style_context ();
+        item_style_context.add_provider (item_style_provider, Gtk.STYLE_PROVIDER_PRIORITY_APPLICATION);
+        item_style_context.add_class (Gtk.STYLE_CLASS_LABEL);
+        camera_options.append (menuitem);
+        int i = (int) camera_options.get_children ().length () - 1;
+        menuitem.activate.connect (() => {
+            request_camera_change (i);
+        });
+        menuitem.show ();
+    }
+
+    public void remove_camera_option (Gst.Device camera) {
+        Gtk.Widget to_remove = null;
+        foreach (var menuitem in camera_options.get_children ()) {
+            var name = (menuitem as Gtk.MenuItem).label;
```

Comment:
The compiler complains about this kind of casting because there is no null check. Since we know that this is for sure a menuitem we can use an unsafe cast
```suggestion
            var name = ((Gtk.MenuItem) menuitem).label;
```